Servers/Infrastructure/OJS.Servers.Infrastructure/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -401,7 +403,21 @@ private static IServiceCollection AddRedis(this IServiceCollection services, ICo
         private static void ConfigureHttpClient(HttpClient client)
             => client.DefaultRequestHeaders.Add(HeaderNames.Accept, MimeTypes.ApplicationJson);
 
-        private static Task UnAuthorizedResponse(RedirectContext<CookieAuthenticationOptions> context)
+        private static Task UnauthorizedResponse(RedirectContext<CookieAuthenticationOptions> context)
+        {
+            context.Response.StatusCode = (int)HttpStatusCode.Unauthorized;
+            context.Response.ContentType = "application/json";
+
+            var result = JsonSerializer.Serialize(new ExceptionResponseModel
+            {
+                Name = "You are not logged in.",
+                Message = "Your session may have expired or you may not be authenticated."
+            });
+
+            return context.Response.WriteAsync(result);
+        }
+
+        private static Task ForbiddenResponse(RedirectContext<CookieAuthenticationOptions> context)
         {
             context.Response.StatusCode = (int)HttpStatusCode.Unauthorized;
```

Comment:
`StatusCode` should be `Forbidden` here.

Reply:
Done.